### DISCUSSION TITLE
How to respond with a Python Requests response?

### DISCUSSION THREAD

**Discussion by ubalklen** [fastapi/fastapi#6009] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import requests

app = FastAPI()


@app.get("/")
async def root():
    r = requests.get("https://google.com")
    return r # Won't work, this is what my question is about
```

### Description

I want my API to query another web app using the standard Python Requests library and give the exact response back to the caller. I know I can create custom responses, but I wonder if there is a built-in way to do that in FastAPI.

If not, any hints on how to create a custom response for my case would be appreciated.

### Operating System

Windows

### FastAPI Version

0.92.0

### Python Version

3.10

**Comment by odiseo0** [comment, marked as answer]
Maybe just 
```python
from fastapi import FastAPI
import requests

app = FastAPI()


@app.get("/")
async def root():
    r = requests.get("https://google.com")
    return r.json()
```

**Reply by ubalklen** [reply]
Thanks,. but I guess this would take care of the content, but not of the headers, right? Response must be identical for my use case.

**Reply by odiseo0** [reply]
Using `Response` and mapping the entire response will do
```python
from fastapi import FastAPI, Response
import requests

app = FastAPI()


@app.get("/")
async def root():
    r = requests.get("https://google.com")
    return Response(r.content, status_code=r.status_code, headers=r.headers) # and so on
